src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.cs

```diff
@@ -9,6 +9,7 @@ namespace System.Collections.Frozen
     public static partial class FrozenDictionary
     {
         public static System.Collections.Frozen.FrozenDictionary<TKey, TValue> ToFrozenDictionary<TKey, TValue>(this System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<TKey, TValue>> source, System.Collections.Generic.IEqualityComparer<TKey>? comparer = null) where TKey : notnull { throw null; }
+        public static System.Collections.Frozen.FrozenDictionary<TKey, TValue> ToFrozenDictionary<TKey, TValue>(this System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<TKey, TValue>> source, System.Collections.Generic.IEqualityComparer<TKey>? comparer, bool optimizeForReading) where TKey : notnull { throw null; }
```

Comment:
Shouldn't the comparer parameter be optional? Seems you need to always supply a comparer when opting for optimized reads, even if null.

Reply:
It can't be optional here or both overloads with optional parameters would be ambiguous when neither are supplied.

Reply:
How about having another overload:
```C#
public static FrozenDictionary<TKey, TValue> ToFrozenDictionary<TKey, TValue>(
    this IEnumerable<KeyValuePair<TKey, TValue>> source, 
    bool optimizeForReading) 
```
Seems many people would want to call this without bothering to specify a converter.

Reply:
> How about having another overload

I'm fine with that if you think it's valuable.  It's a pure addition so we can merge this and follow-up on adding more overloads.